repository/src/main/java/org/apache/atlas/tasks/AtlasTaskService.java

```diff
@@ -71,4 +70,30 @@ public TaskSearchResult getTasks(TaskSearchParams searchParams) throws AtlasBase
 
         return ret;
     }
+
+    @Override
+    public void retryTask(String taskGuid) throws AtlasBaseException {
+        AtlasGraphQuery query = graph.query()
+                .has(Constants.TASK_TYPE_PROPERTY_KEY, Constants.TASK_TYPE_NAME)
+                .has(TASK_GUID, taskGuid);
+
+        Iterator<AtlasVertex> results = query.vertices().iterator();
+
+        if (results.hasNext()) {
+            AtlasVertex atlasVertex = results.next();
+
+            String status = atlasVertex.getProperty(Constants.TASK_STATUS, String.class);
+
+            if (status.equals(AtlasTask.Status.PENDING.toString()) || status.equals(AtlasTask.Status.IN_PROGRESS.toString())) {
```

Comment:
do we need to check for status COMPLETE?

why not just check if status not equals to FAILED?

Reply:
Yes, mentioned this in check-in before. Changed the behavior so that this can be applied already completed, failed and deleted tasks. Sometimes we would need to retry the completed task for testing purposes

Reply:
ok I see

Can we add a comment for this and give some context for the reader of the code? think it is not intuitive

Reply:
sure definitely

Reply:
addressed